packages/frontend/src/redux/slices/account/createAccountThunks.js

```diff
@@ -187,3 +191,31 @@ export const finishSetupImplicitAccount = createAsyncThunk(
         await dispatch(addLocalKeyAndFinishSetup({ accountId: implicitAccountId, recoveryMethod, publicKey })).unwrap();
     }
 );
+
+export const finishLocalSetupForZeroBalanceAccount = createAsyncThunk(
+    `${SLICE_NAME}/finishLocalSetupForZeroBalanceAccount`,
+    async ({
+        implicitAccountId,
+        recoveryMethod
+    }, { dispatch }) => {
+        try {
+            if (recoveryMethod === 'ledger') {
+                const ledgerHDPath = getLedgerHDPath(implicitAccountId);
+                await dispatch(signInWithLedger({ path: ledgerHDPath })).unwrap();
+            } else {
+                const account = await wallet.getAccount(implicitAccountId);
+                const accessKeys = await account.getAccessKeys();
+                const fullAccessKeys = accessKeys.filter((it) => it.access_key && it.access_key.permission === 'FullAccess');
```

Comment:
```suggestion
                const fullAccessKeys = accessKeys.filter((it) => it.access_key?.permission === 'FullAccess');
```
nit: optional chaining 😄